File: app/android/app/src/main/java/bosch/smartcampus/thermalcomfortstudy/packet/ActivityDescriptionItem.java
package bosch.smartcampus.thermalcomfortstudy.packet;

import bosch.smartcampus.thermalcomfortstudy.lib.Timestamp;

/**
 * Created by rsukkerd on 6/27/16.
 */
public class ActivityDescriptionItem extends SurveyDataItem {
    private static final String ACTIVITY_DESCRIPTION_NAME = "ActivityDescription";

    private String description;

    public ActivityDescriptionItem(String username, Timestamp timestamp,
                                   String description) {
        super(username, timestamp);
        this.description = description;
    }

    public String getActivityDescription() {
        return description;
    }

    @Override
    public String toXML() {
        StringBuilder builder = new StringBuilder();
        builder.append("<" + ELEMENT + " id=\"" + getId() + "\">");
        builder.append(getTransducerDataElement(ACTIVITY_DESCRIPTION_NAME, description));
        builder.append("</" + ELEMENT + ">");
        return builder.toString();
    }
}

File: app/iOS/Thermal Comfort Study/ActivityDescriptionItem.swift
//
//  ActivityDescriptionItem.swift
//  Thermal Comfort Study
//
//  Created by Madhav Achar on 10/21/16.
//  Copyright © 2016 boschsc. All rights reserved.
//

import Foundation

public class ActivityDescriptionItem: SurveyDataItem {
    private static let ACTIVITY_DESCRIPTION_NAME = "ActivityDescription"
    
    init(username: String, timestamp: Date, description: String) {
        super.init(username, timestamp, type(of: self).ACTIVITY_DESCRIPTION_NAME, description)
    }
}
